subsys/logging/Kconfig

```diff
@@ -209,11 +209,37 @@ config LOG_INPLACE_PROCESS
 	bool "Enable in place processing"
 	help
 	  When enabled log is processed in the context of the call. It impacts
-	  performance of the system since time consuming operations are performed in
-	  the context of the log entry (e.g. high priority interrupt). When enabled
-	  LOG_BUFFER_SIZE can be reduced. Logger backends must support exclusive
-	  access to work flawlessly in that mode because one log operation can be
-	  interrupted by another one in higher priority context.
+	  performance of the system since time consuming operations are
+	  performed in the context of the log entry (e.g. high priority
+	  interrupt). When enabled LOG_BUFFER_SIZE can be reduced. Logger
+	  backends must support exclusive access to work flawlessly in that mode
+	  because one log operation can be interrupted by another one in higher
+	  priority context.
+
+config LOG_PROCESS_TRIGGER_THRESHOLD
+	int "Amount of buffered logs which triggers processing thread."
+	default 0
+	help
+	  When number of buffered messages reaches the threshold thread is waken
+	  up. Log processing thread ID is provided during log initialization.
+	  Set 0 to disable the feature.
+
+config LOG_PROCESS_THREAD
```

Comment:
I suggest 2 different options:

LOG_PROCESS_THREAD: Allow for log processing thread, enables other LOG_PROCESS_THREAD_* options.
LOG_PROCESS_THREAD_INTERNAL: Enables internal thread for log processing.